File: result/translate/starchat/java2python/python/MAXIMUM_POSSIBLE_SUM_WINDOW_ARRAY_ELEMENTS_WINDOW_ARRAY_UNIQUE.py
def f_gold(A, B, n):
        mp = set()
        result = 0
        curr_sum = 0
        curr_begin = 0
        for i in range(n):
            while A[curr_begin] in mp:
                mp.remove(A[curr_begin])
                curr_sum -= B[curr_begin]
                curr_begin += 1
            mp.add(A[i])
            curr_sum += B[i]
            result = max(result, curr_sum)
        return result

File: result/translate/starchat/python2java/java/MAXIMUM_POSSIBLE_SUM_WINDOW_ARRAY_ELEMENTS_WINDOW_ARRAY_UNIQUE.java
public static int f_gold ( int A [ ], int B [ ], int n ) {
        Set < Integer > mp = new HashSet < > ( );
        int result = 0, curr_sum = 0, curr_begin = 0;
        for ( int i = 0 ; i < n ; i++ ) {
            while ( A [ i ]!= mp.iterator ( ) ) {
                mp.remove ( A [ curr_begin ] );
                curr_sum -= B [ curr_begin ];
                curr_begin++;
            }
            mp.add ( A [ i ] );
            curr_sum += B [ i ];
            result = Math.max ( result, curr_sum );
        }
        return result;
    }